File: short_problems/Java/LinearSearch.java
import java.util.Arrays;
import java.util.Collection;

public class LinearSearch {
    public static <T> boolean linearSearch(Collection<? extends T> collection, T key) {
        for (T it : collection) {
            if (it.equals(key)) {
                return true;
            }
        }

        return false;
    }
    public static void main(String[] args) {
        final Collection<Integer> coll1 = Arrays.asList(1, 4, 2, 5, 8, -1);
        final int key1 = 2;
        System.out.println(linearSearch(coll1, key1));
        final int key2 = 3;
        System.out.println(linearSearch(coll1, key2));
    }
}

File: short_problems/Python/linear_search.py
#!/usr/local/bin/python3

def linear_search(collection, key):
    return key in collection

if __name__ == "__main__":
    coll1 = [7, 2, 1, 5, 4, 3, -1]
    key1 = 1
    print(linear_search(coll1, key1))
    key1 = 6
    print(linear_search(coll1, key1))
